src/sentry/runner/commands/notifications.py

```diff
@@ -0,0 +1,119 @@
+
+
+@send.command("email")
+@click.option(
+    "-s",
+    "--source",
+    help="Registered template source (see `sentry notifications list`)",
+    default="error-alert-service",
+)
+@click.option("-t", "--target", help="Recipient email address", default="user@example.com")
+def send_email(source: str, target: str) -> None:
+    """
+    Send an email notification.
+    Note: Requires configuring SMTP settings in .sentry/config.yml.
+    """
+    from sentry import options
+    from sentry.runner import configure
+
+    configure()
+
+    if options.get("mail.backend") in {"dummy", "console"} or any(
+        options.get(key) is None
+        for key in ["mail.host", "mail.port", "mail.username", "mail.password"]
+    ):
+        click.echo("Unable to send email with current configuration!")
+        click.echo(
+            """Please update .sentry/config.yml:
+  - remove `mail.backend` (if set to 'dummy' or 'console')
+  - set `mail.host`, `mail.port`, `mail.username` and `mail.password`"""
+        )
+        return
+
+    from sentry.notifications.platform.registry import template_registry
+    from sentry.notifications.platform.service import NotificationService
+    from sentry.notifications.platform.target import GenericNotificationTarget
+    from sentry.notifications.platform.types import (
+        NotificationProviderKey,
+        NotificationTargetResourceType,
+    )
+
+    email_target = GenericNotificationTarget(
+        provider_key=NotificationProviderKey.EMAIL,
+        resource_type=NotificationTargetResourceType.EMAIL,
+        resource_id=target,
+    )
+    template_cls = template_registry.get(source)
+    NotificationService(data=template_cls.example_data).notify(targets=[email_target])
+    click.echo(f"Example '{source}' email sent to {target}.")
+
+
+@send.command("slack")
+def send_slack() -> None:
+    """
+    Send a Slack notification
+    """
+    click.echo("Not implemented yet!")
+
+
+@send.command("msteams")
+def send_msteams() -> None:
+    """
+    Send a Microsoft Teams notification.
+    """
+    click.echo("Not implemented yet!")
+
+
+@send.command("discord")
+def send_discord() -> None:
+    """
+    Send a Discord notification.
+    """
+    click.echo("Not implemented yet!")
+
+
+@notifications.command("list")
+def list() -> None:
+    """
+    Lists registered notification data.
+    """
+    from sentry.runner import configure
+
+    configure()
+
+    from sentry.notifications.platform.registry import provider_registry, template_registry
+
+    click.echo("\nRegistered notification providers:")
+    for provider_key, provider_cls in provider_registry.registrations.items():
+        click.echo(f"• key: {provider_key}, class: {provider_cls.__name__}")
+
+    click.echo("\nRegistered notification templates:")
+    # XXX: For some reason, using a defaultdict(list) here causes the command to interpret
+    # the .append() as new arguments, causing an early exit and error.
+    category_to_sources = {}
```

Comment:
I don't understand the issue here at all, but this was able to stop it from breaking.

```
> $ sentry notifications list
Registered notification providers:
• key: discord, class: DiscordNotificationProvider
• key: email, class: EmailNotificationProvider
• key: msteams, class: MSTeamsNotificationProvider
• key: slack, class: SlackNotificationProvider

Registered notification templates:
Usage: python -m sentry [OPTIONS]
Try 'python -m sentry --help' for help.

Error: Got unexpected extra arguments (notifications list)
```
This is the error that'd be produced when using a defaultdict. I think it has something to do with automatically creating the list when it hits a key error, but I'm not sure why that empty list is being eaten by the function as a new argument.